From 20c1ba33c66f04751da5a1470251d8fd33af0363 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 30 Dec 2021 17:42:49 +0200
Subject: [PATCH] [SW-68669] efuse: write address range as part of init

Components that are executing before preboot/ppboot might change the
default address range of the efuse reading.
In order to assure that all the addresses being fetched during the
efuse read command, the default address range should be restored
prior to the read command execution.

The default range value of 0x3F80 is suitable for all the current
platforms that are using efuse (H3, H5, H6)

Change-Id: Ic2d4d8aaf38b3b356790e5be91664d7529ee1f18
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/lib/common/efuse.c b/lib/common/efuse.c
index 80118e7..303fbc7 100644
--- a/lib/common/efuse.c
+++ b/lib/common/efuse.c
@@ -25,6 +25,7 @@
 #define FUSE_TIMEOUT_US		100000
 #define FUSE_DELAY_US		20
 #define FUSE_RETRIES		(FUSE_TIMEOUT_US / FUSE_DELAY_US)
+#define FUSE_ALL_ADDRESSES	0x3F80
 
 static int efuse_init;
 
@@ -36,6 +37,8 @@
 	if (efuse_init == 1)
 		return -1;
 
+	regmap_writel(FUSE_ALL_ADDRESSES, mmPSOC_EFUSE_READ_ADDRESS);
+
 	read_reg = get_efuse_read_reg();
 
 	regmap_writel(1, read_reg);
